internal/venice-common/src/test/java/com/linkedin/venice/blobtransfer/client/TestP2PFileTransferClientHandler.java

```diff
@@ -0,0 +1,208 @@
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.CharsetUtil;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class TestP2PFileTransferClientHandler {
+  EmbeddedChannel ch;
+  Path baseDir;
+  String TEST_STORE = "test_store";
+  int TEST_VERSION = 1;
+  int TEST_PARTITION = 0;
+  CompletionStage<InputStream> inputStreamFuture;
+
+  @BeforeMethod
+  public void setUp() throws IOException {
+    baseDir = Files.createTempDirectory("tmp");
+    inputStreamFuture = new CompletableFuture<>();
+    ch = new EmbeddedChannel(
+        new P2PFileTransferClientHandler(
+            baseDir.toString(),
+            inputStreamFuture,
+            TEST_STORE,
+            TEST_VERSION,
+            TEST_PARTITION));
+  }
+
+  @AfterMethod
+  public void teardown() throws IOException {
+    ch.close();
+    Files.walk(baseDir).sorted(Comparator.reverseOrder()).forEach(path -> {
+      try {
+        Files.delete(path);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  @Test
+  public void testFailToGetResponse() {
+    DefaultHttpResponse response =
+        new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    ch.writeInbound(response);
+    try {
+      inputStreamFuture.toCompletableFuture().get();
+      Assert.fail("Expected exception not thrown");
+    } catch (Exception e) {
+      Assert.assertTrue(e.getCause() instanceof VeniceException);
+      Assert.assertEquals(
+          e.getCause().getMessage(),
+          "Failed to fetch file from remote peer. Response: 500 Internal Server Error");
+    }
+  }
+
+  @Test
+  public void testInvalidResponseHeader() {
+    DefaultHttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+    ch.writeInbound(response);
+    try {
+      inputStreamFuture.toCompletableFuture().get();
+      Assert.fail("Expected exception not thrown");
+    } catch (Exception e) {
+      Assert.assertTrue(e.getCause() instanceof VeniceException);
+      Assert.assertEquals(e.getCause().getMessage(), "No file name specified in the response for test_store_v1-0");
+    }
+  }
+
+  @Test
+  public void testFileSizeMismatchTransfer() {
+    // response
+    DefaultHttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
```

Comment:
can we re-write comment,  ie. `creates http response with expected content length of 5`

Reply:
same w/ other tests